src/Generators/Generator.php

```diff
@@ -206,39 +184,52 @@ public function getDescription()
      * ]);
      * ~~~
      */
-    public function rules()
+    public function rules(): array
     {
         return [
-            [['template'], 'required', 'message' => 'A code template must be selected.'],
-            [['template'], 'validateTemplate'],
+            'template' => [
+                (new Required())->message('A code template must be selected.')
+            ],
         ];
     }
 
     /**
      * Loads sticky attributes from an internal file and populates them into the generator.
      * @internal
      */
-    public function loadStickyAttributes()
+    public function loadStickyAttributes(): void
     {
         $stickyAttributes = $this->stickyAttributes();
         $path = $this->getStickyDataFile();
         if (is_file($path)) {
-            $result = json_decode(file_get_contents($path), true);
+            $result = Json::decode(file_get_contents($path), true);
             if (is_array($result)) {
                 foreach ($stickyAttributes as $name) {
-                    if (isset($result[$name])) {
+                    if (isset($result[$name]) && property_exists($this, $name)) {
                         $this->$name = $result[$name];
```

Comment:
I think it can cause a lot of problems. For example, I can replace `$aliases` variable :(
I think it should be redesigned

Reply:
Let's do it in the next PRs ;)